src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -1802,6 +1802,31 @@ private uint getFieldOffset(CORINFO_FIELD_STRUCT_* field)
         private bool isWriteBarrierHelperRequired(CORINFO_FIELD_STRUCT_* field)
         { throw new NotImplementedException("isWriteBarrierHelperRequired"); }
 
+        private CORINFO_FIELD_ACCESSOR getFieldIntrinsic(FieldDesc field)
+        {
+            var owningType = field.OwningType;
+            if ((owningType.IsWellKnownType(WellKnownType.IntPtr) ||
+                    owningType.IsWellKnownType(WellKnownType.UIntPtr)) &&
+                        field.Name == "Zero")
+            {
+                return CORINFO_FIELD_ACCESSOR.CORINFO_FIELD_INTRINSIC_ZERO;
+            }
+            // Disabled outside READYTORUN because of https://github.com/dotnet/corert/issues/6601
+#if READYTORUN
+            else if (owningType.IsString && field.Name == "Empty")
+            {
+                return CORINFO_FIELD_ACCESSOR.CORINFO_FIELD_INTRINSIC_EMPTY_STRING;
+            }
+#endif
+            else if (owningType.Name == "BitConverter" && owningType.Namespace == "System" &&
```

Comment:
Getting the name every time through here won't be cheap. Should we have recognize&cache IntrisicAttribute on fields, like we are doing it for methods?

Reply:
I kind of knew you won't let me cut corners here, but I had to try :). I'll take a look at that.

Reply:
Done. I've also submitted dotnet/coreclr#21162 so that we cover this for CPAOT in one go.